pkg/cmd/image/remove.go

```diff
@@ -89,10 +92,27 @@ func Remove(ctx context.Context, client *containerd.Client, args []string, optio
 		},
 	}
 
-	err = walker.WalkAll(ctx, args, true)
-	if err != nil && options.Force {
-		logrus.Error(err)
-		return nil
+	var errs []string
+	var fatalErr bool
+	for _, req := range args {
+		n, err := walker.Walk(ctx, req)
+		if err != nil {
+			fatalErr = true
+		}
+		if err == nil && n == 0 {
+			err = fmt.Errorf("no such image: %s", req)
+		}
+		if err != nil {
+			errs = append(errs, err.Error())
+		}
+	}
+
+	if len(errs) > 0 {
+		msg := fmt.Sprintf("%d errors:\n%s", len(errs), strings.Join(errs, "\n"))
+		if !options.Force || fatalErr {
+			return errors.New(msg)
+		}
+		logrus.Error(msg)
```

Comment:
This is to exactly match docker's [error handling behavior](https://github.com/docker/cli/blob/dfb36eaef8729914749500b148649e05ceeddd03/cli/command/image/remove.go#L61-L88) (for each err, if not `IsErrNotFound`, then it's fatal; if there is any fatal error, return a joined error, else log error and return nil).

If using `WalkAll`, an edge case will be inconsistent with docker: if all errors are `IsErrNotFound` (e.g. `docker|nerdctl rmi -f afsd afdafaf aff`), docker return nil and nerdctl return a joined err. TBH I don't feel it makes much sense to distinguish the specific edge case in docker...